packages/mgt-components/src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -936,6 +931,11 @@ export class MgtPeoplePicker extends MgtTemplatedTaskComponent {
     let people = this.people;
     const input = this.userInput.toLowerCase();
     const provider = Providers.globalProvider;
+    this._hasMaxSelections =
+      this.selectionMode === 'single' &&
+      (this.selectedPeople.length >= 1 ||
+        this.defaultSelectedUserIds.length >= 1 ||
+        this.defaultSelectedGroupIds.length >= 1);
```

Comment:
This still has the same defect as the previous version in that if a non-existent userId is passed then we can incorrectly disable the input. 

Just revert to having the prior implementation of get hasMaxSelections and now the check on 1056 will disable the input as required